anydex/wallet/stellar/xlm_db.py

```diff
@@ -0,0 +1,240 @@
+        """
+        pending_txs = self.session.query(Transaction).filter(Transaction.is_pending.is_(True)).all()
+        confirmed_txs = self.session.query(Transaction).filter(Transaction.is_pending.is_(False)).all()
+        for transaction in transactions:
+            if transaction in pending_txs:
+                self._update_transaction(transaction)
+            elif transaction not in confirmed_txs:
+                self._insert_transaction(transaction)
+        self.session.commit()
+
+    def _update_transaction(self, transaction):
+        """
+        Update a pending transaction and it's corresponding payments
+        :param transaction: transaction to update
+        """
+        self.session.query(Transaction).filter(Transaction.hash == transaction.hash).update({
+            Transaction.is_pending: False,
+            Transaction.succeeded: transaction.succeeded,
+            Transaction.date_time: transaction.date_time
+        })
+
+    def _insert_transaction(self, transaction):
+        """
+        Insert a transactions into the database.
+        Todo what about merge account
+        The payments (payment and create account) will also be added from the transactions.
+
+        :param transaction: transaction to insert
+        """
+        self.session.add(transaction)
+
+        xdr_unpacker = StellarXDRUnpacker(b64decode(transaction.transaction_envelope))
+        operations = xdr_unpacker.unpack_TransactionEnvelope().tx.operations
+
+        for operation in operations:
+            source_account = operation.sourceAccount
+            # check if the operation has a source account
+            if not source_account:
+                source_account = transaction.source_account
+
+            else:
+                source_account = Keypair.from_raw_ed25519_public_key(source_account[0].ed25519).public_key
+            body = operation.body
+            # we only care about create account and payment for the time being
+            if body.type == 0:  # create account
+                create_account_op = body.createAccountOp
+                payment = Payment(
+                    amount=create_account_op.startingBalance,
+                    asset_type="native",
+                    transaction_hash=transaction.hash,
+                    to=Keypair.from_raw_ed25519_public_key(create_account_op.destination.ed25519).public_key,
+                    from_=source_account
+                )
+            elif body.type == 1:  # payment
+                payment_op = body.paymentOp
+                payment = Payment(amount=payment_op.amount,
+                                  asset_type="native",
+                                  transaction_hash=transaction.hash,
+                                  to=Keypair.from_raw_ed25519_public_key(payment_op.destination.ed25519).public_key,
+                                  from_=source_account)
+            self.session.add(payment)
```

Comment:
Is there an _else_ possibility, in which case `self.session.add(payment)` would result in providing no defined variable?

Reply:
Yeah, your right. Nice catch